src/node.cc

```diff
@@ -1340,41 +1335,22 @@ Local<Value> MakeCallback(Environment* env,
     }
   }
 
-  if (ran_init_callback && !pre_fn.IsEmpty()) {
-    TryCatch try_catch(env->isolate());
-    MaybeLocal<Value> ar = pre_fn->Call(env->context(), object, 0, nullptr);
-    if (ar.IsEmpty()) {
-      ClearFatalExceptionHandlers(env);
-      FatalException(env->isolate(), try_catch);
-      return Local<Value>();
-    }
-  }
+  // TODO(trevnorris): Correct this once node::MakeCallback() support id and
+  // triggerId. Consider completely removing it until then so the async id can
+  // propagate through to the fatalException after hook calls.
+  AsyncHooks::ExecScope exec_scope(env, 0, 0);
```

Comment:
Basically, what I said in IRC – the code runs just fine, but it won’t give you the right async id for that (because the default `MakeCallback` signature doesn’t allow specifying that).